service/security/shiro/src/main/java/org/eclipse/kapua/service/authentication/shiro/realm/JwtCredentialsHandler.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2022 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Eurotech - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.kapua.service.authentication.shiro.realm;
+
+import com.google.common.base.Strings;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.shiro.authc.AuthenticationToken;
+import org.eclipse.kapua.service.authentication.JwtCredentials;
+import org.eclipse.kapua.service.authentication.LoginCredentials;
+import org.eclipse.kapua.service.authentication.exception.KapuaAuthenticationErrorCodes;
+import org.eclipse.kapua.service.authentication.exception.KapuaAuthenticationException;
+import org.eclipse.kapua.service.authentication.shiro.JwtCredentialsImpl;
+
+import java.util.Optional;
+
+/**
+ * {@link JwtCredentials} {@link LoginCredentialsHandler} implementation.
+ *
+ * @since 2.0.0
+ */
+public class JwtCredentialsHandler implements LoginCredentialsHandler {
+
+    @Override
+    public boolean canProcess(LoginCredentials loginCredentials) {
+        return loginCredentials instanceof JwtCredentials;
+    }
+
+    @Override
+    public ImmutablePair<AuthenticationToken, Optional<String>> mapToShiro(LoginCredentials loginCredentials) throws KapuaAuthenticationException {
+        JwtCredentialsImpl jwtCredentials = JwtCredentialsImpl.parse((JwtCredentials) loginCredentials);
+        final String openIDidToken = Optional.ofNullable(jwtCredentials.getIdToken())
+                .filter(token -> !Strings.isNullOrEmpty(token))
+                .orElseThrow(() -> new KapuaAuthenticationException(KapuaAuthenticationErrorCodes.INVALID_LOGIN_CREDENTIALS));
+
+        return ImmutablePair.of(jwtCredentials, Optional.of(openIDidToken));
```

Comment:
That one comes out from how the SSO session specific task are handled. 

It would be much nicer if the KapuaSession is created from the CredentialHelper, so we can have different many types of Session according to the credentials given. This for sure has its own challenges, but I think it is outside of the scope of this PR

Reply:
See the solution proposed in https://github.com/Coduz/kapua/pull/11 - subclassing AuthenticationToken to provide the additional "openIdToken" property is cleaner, because each implementation can specify whether such token CAN be provided or is a fixed-value Optional.empty()

That said, should the need to create different KapuaSession based on the given credentials (to the point that no common logic can be found, I mean) would definitely justify the approach you suggest - let's burn that bridge when we get there